services/collaboration/pkg/connector/fileconnector.go

```diff
@@ -1175,18 +1180,16 @@ func (f *FileConnector) CheckFileInfo(ctx context.Context) (*ConnectorResponse,
 	}
 
 	breadcrumbFolderName := path.Dir(statRes.Info.Path)
-	if breadcrumbFolderName == "." || breadcrumbFolderName == "" {
+	if breadcrumbFolderName == "." || breadcrumbFolderName == "" || breadcrumbFolderName == "/" {
 		breadcrumbFolderName = statRes.GetInfo().GetSpace().GetName()
 	}
 
-	ocisUrl, err := url.Parse(f.cfg.Commons.OcisURL)
+	ocisURL, err := url.Parse(f.cfg.Commons.OcisURL)
 	if err != nil {
 		return nil, err
 	}
-	breadcrumbFolderURL, viewAppUrl, editAppUrl := *ocisUrl, *ocisUrl, *ocisUrl
-	breadcrumbFolderURL.Path = path.Join(breadcrumbFolderURL.Path, "f", storagespace.FormatResourceID(statRes.GetInfo().GetId()))
-	viewAppUrl.Path = path.Join(viewAppUrl.Path, "external"+strings.ToLower(f.cfg.App.Name))
-	editAppUrl.Path = path.Join(editAppUrl.Path, "external"+strings.ToLower(f.cfg.App.Name))
+	privateLinkURL := *ocisURL
```

Comment:
maybe something like:
```
privateLinkURL := &url.URL{}
*privateLinkURL = *ocisURL  // shallow copy the data
privateLinkURL.Path = .....
```
It might be clearer since most of the variables are pointers.